### PR TITLE
Fixed errors with existing branches and changed names of branches to …

### LINKED ISSUE #114: branch name should include full library id
Currently branch name contains only artifact id. Many libraries have generic artifact name like "interface" or "driver" and group is significant there. We may face conflicts if we don't use the group + artifact for the name.

### PR DESCRIPTION
**Pull request by mikkoziel**
Fixed errors with existing branches and changed names of branches to include group name.
Fixes #114 and #74

### CODE REVIEW

File: core/src/main/kotlin/org/virtuslab/bazelsteward/core/common/GitOperations.kt [hunk 1]
```diff
@@ -13,7 +13,15 @@ class GitOperations(private val config: Config) {
 
   suspend fun createBranchWithChange(change: FileUpdateSearch.FileChangeSuggestion): GitBranch {
     val branch = fileChangeSuggestionToBranch(change)
-    git.checkout(branch.name, true)
+    try {
+      git.checkout(branch.name, true)
+    } catch (e: RuntimeException) {
+      if (e.message!!.contains(branch.name)) {
```
Comment (lukaszwawrzyk) [inline core/src/main/kotlin/org/virtuslab/bazelsteward/core/common/GitOperations.kt:19]: if message is nullable, why risking an NPE? e.message?.contains(branch.name) == true

File: core/src/main/kotlin/org/virtuslab/bazelsteward/core/common/GitOperations.kt [hunk 2]
```diff
@@ -13,7 +13,15 @@ class GitOperations(private val config: Config) {
...
+        git.deleteBranch(branch.name)
+        git.checkout(branch.name, true)
```
Comment (lukaszwawrzyk) [inline core/src/main/kotlin/org/virtuslab/bazelsteward/core/common/GitOperations.kt:21]: use named parameter for this `true`

File: maven/src/main/kotlin/org/virtuslab/bazelsteward/maven/MavenCoordinates.kt [hunk 3]
```diff
@@ -12,6 +12,9 @@ data class MavenLibraryId(val group: String, val artifact: String) : LibraryId {
 
   override val name: String
     get() = artifact
+
```
Comment (lukaszwawrzyk) [inline maven/src/main/kotlin/org/virtuslab/bazelsteward/maven/MavenCoordinates.kt:15]: This should be the other way around. Do not touch library id, it has good api.

In MavenLibraryId, name should be implemented as "$group:artifact".
Imagine other build systems with a different set of components that define a library. We just want a one string at the end.

Comment (lukaszwawrzyk) [reply maven/src/main/kotlin/org/virtuslab/bazelsteward/maven/MavenCoordinates.kt:15]: In GitBranch, you should just sanitize the name. For example replace ":" with a "/".
In general we might replace all illegal characters with something (or remove them), but this is for another task.

**Review by lukaszwawrzyk — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/org/virtuslab/bazelsteward/core/common/GitOperations.kt (hunk 2 above)
Comment (mikkoziel) [reply core/src/main/kotlin/org/virtuslab/bazelsteward/core/common/GitOperations.kt:21]: Fixed

File: maven/src/main/kotlin/org/virtuslab/bazelsteward/maven/MavenCoordinates.kt (hunk 3 above)
Comment (mikkoziel) [reply maven/src/main/kotlin/org/virtuslab/bazelsteward/maven/MavenCoordinates.kt:15]: Changed the logic

File: core/src/main/kotlin/org/virtuslab/bazelsteward/core/common/GitOperations.kt (hunk 1 above)
Comment (mikkoziel) [reply core/src/main/kotlin/org/virtuslab/bazelsteward/core/common/GitOperations.kt]: fixed

**Review by lukaszwawrzyk — APPROVED** [review]

**Pull request merged**